modules/saku-policy-store/src/saku/dal_datalevin.clj

```diff
@@ -52,38 +41,34 @@
 
 (defmethod interface/get-resource-policies :datalevin [obj db-or-drns & [drns]]
   (let [[db' drns'] (if drns [db-or-drns drns] [(interface/db obj) db-or-drns])]
-    (impl-get-resource-policies db' drns')))
+    (impl-get-*-policies db' drns' :statement/identities)))
 
 
 (defmethod interface/get-identity-policies :datalevin [obj db-or-drns & [drns]]
   (let [[db' drns'] (if drns [db-or-drns drns] [(interface/db obj) db-or-drns])]
-    (impl-get-identity-policies db' drns')))
+    (impl-get-*-policies db' drns' :statement/resources)))
 
 
+(defn upsert-*-policies
+  [db-conn policies]
+  (let [tx-data (into []
+                  (mapcat (fn [{:keys [policy/drn] :as policy}]
+                            [[:db/retract [:policy/drn drn] :policy/statements]
+                             policy]))
+                  policies)]
```

Comment:
I always prefer an `map`-esque function over `reduce` since I've found `reduce` will result in more nuanced bugs since it's up to you to manage the accumulator.